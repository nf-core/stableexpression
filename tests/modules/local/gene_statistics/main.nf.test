nextflow_process {

    name "Test Process GENE_STATISTICS"
    script "modules/local/gene_statistics/main.nf"
    process "GENE_STATISTICS"
    tag "genestats"
    tag "module"

    test("Without variation data") {

        when {

            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/test_data/merge_counts/output/all_counts.parquet', checkIfExists: true)
                ch_metadata = Channel.fromPath( '$projectDir/tests/test_data/gene_statistics/input/metadata*.csv', checkIfExists: true ).collect()
                ch_mapping = Channel.fromPath( '$projectDir/tests/test_data/gene_statistics/input/mapping*.csv', checkIfExists: true ).collect()
                ch_m_measures = Channel.of( 'none' )
                input[0] = ch_counts
                input[1] = ch_metadata
                input[2] = ch_mapping
                input[3] = ch_m_measures
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("With m-measures") {

        tag "genestats_m_measure"

        when {

            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/test_data/merge_counts/output/all_counts.parquet', checkIfExists: true)
                ch_metadata = Channel.fromPath( '$projectDir/tests/test_data/gene_statistics/input/metadata*.csv', checkIfExists: true ).collect()
                ch_mapping = Channel.fromPath( '$projectDir/tests/test_data/gene_statistics/input/mapping*.csv', checkIfExists: true ).collect()
                ch_m_measures = Channel.fromPath( '$projectDir/tests/test_data/gene_statistics/input/m_measures.csv', checkIfExists: true )
                input[0] = ch_counts
                input[1] = ch_metadata
                input[2] = ch_mapping
                input[3] = ch_m_measures
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
