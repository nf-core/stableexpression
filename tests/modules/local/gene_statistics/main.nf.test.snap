{
    "Small dataset": {
        "content": [
            {
                "0": [
                    "stats_all_genes.csv:md5,3dd505bec3381b67da2dd316b49e2727"
                ],
                "1": [
                    "stats_most_stable_genes.csv:md5,c07e07ae7d28ccfb63a2fc88f05f77a1"
                ],
                "2": [
                    "log_counts.csv:md5,676b63d29a0658c9c1766021d3d8f862"
                ],
                "3": [
                    [
                        "GENE_STATISTICS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "4": [
                    [
                        "GENE_STATISTICS",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "log_count_summary": [
                    "log_counts.csv:md5,676b63d29a0658c9c1766021d3d8f862"
                ],
                "stats_all_genes": [
                    "stats_all_genes.csv:md5,3dd505bec3381b67da2dd316b49e2727"
                ],
                "stats_most_stable_genes": [
                    "stats_most_stable_genes.csv:md5,c07e07ae7d28ccfb63a2fc88f05f77a1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-15T09:24:07.245081166"
    },
    "With m-measures": {
        "content": [
            {
                "0": [
                    "top_stable_genes_summary.csv:md5,d6a285243f39f0c03110f3d94af410f7"
                ],
                "1": [
                    "stats_all_genes.csv:md5,ffb5e00437d554df44b3f015057b0cd4"
                ],
                "2": [
                    "all_log_counts.csv:md5,2450ea9cdb200c2c2623b5e3f95d91aa"
                ],
                "3": [
                    "top_stable_genes_transposed_log_counts.csv:md5,71504afee585adbb9a0051ac8a93d041"
                ],
                "4": [
                    [
                        "GENE_STATISTICS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "5": [
                    [
                        "GENE_STATISTICS",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "all_statistics": [
                    "stats_all_genes.csv:md5,ffb5e00437d554df44b3f015057b0cd4"
                ],
                "log_counts": [
                    "all_log_counts.csv:md5,2450ea9cdb200c2c2623b5e3f95d91aa"
                ],
                "top_stable_genes_summary": [
                    "top_stable_genes_summary.csv:md5,d6a285243f39f0c03110f3d94af410f7"
                ],
                "top_stable_genes_transposed_log_counts": [
                    "top_stable_genes_transposed_log_counts.csv:md5,71504afee585adbb9a0051ac8a93d041"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-29T16:29:47.579297076"
    },
    "Without variation data": {
        "content": [
            {
                "0": [
                    "top_stable_genes_summary.csv:md5,ad164f68abf18407a1450350be1ca8d1"
                ],
                "1": [
                    "stats_all_genes.csv:md5,d6c2474026e3650405ce70cdf94dff16"
                ],
                "2": [
                    "all_log_counts.csv:md5,2450ea9cdb200c2c2623b5e3f95d91aa"
                ],
                "3": [
                    "top_stable_genes_transposed_log_counts.csv:md5,71504afee585adbb9a0051ac8a93d041"
                ],
                "4": [
                    [
                        "GENE_STATISTICS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "5": [
                    [
                        "GENE_STATISTICS",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "all_statistics": [
                    "stats_all_genes.csv:md5,d6c2474026e3650405ce70cdf94dff16"
                ],
                "log_counts": [
                    "all_log_counts.csv:md5,2450ea9cdb200c2c2623b5e3f95d91aa"
                ],
                "top_stable_genes_summary": [
                    "top_stable_genes_summary.csv:md5,ad164f68abf18407a1450350be1ca8d1"
                ],
                "top_stable_genes_transposed_log_counts": [
                    "top_stable_genes_transposed_log_counts.csv:md5,71504afee585adbb9a0051ac8a93d041"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-29T15:58:15.518228873"
    }
}